tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py

```diff
@@ -1139,6 +1139,37 @@ def test_describe_stack_pagination():
     assert "NextToken" not in resp2.keys()
 
 
+@mock_cloudformation
+def test_describe_stack_resource():
+    cf_conn = boto3.client("cloudformation", region_name="us-east-1")
+    cf_conn.create_stack(StackName="test_stack", TemplateBody=dummy_template_json)
+
+    stack = cf_conn.describe_stacks(StackName="test_stack")["Stacks"][0]
+
+    response = cf_conn.describe_stack_resource(
+        StackName=stack["StackName"], LogicalResourceId="EC2Instance1"
+    )
+
+    resource = response["StackResourceDetail"]
+    resource["LogicalResourceId"].should.equal("EC2Instance1")
+    resource["ResourceStatus"].should.equal("CREATE_COMPLETE")
+    resource["ResourceType"].should.equal("AWS::EC2::Instance")
+    resource["StackId"].should.equal(stack["StackId"])
+
+
+@mock_cloudformation
+def test_describe_stack_resource_when_resource_does_not_exist():
+    cf_conn = boto3.client("cloudformation", region_name="us-east-1")
+    cf_conn.create_stack(StackName="test_stack", TemplateBody=dummy_template_json)
+
+    stack = cf_conn.describe_stacks(StackName="test_stack")["Stacks"][0]
+
+    with pytest.raises(ClientError, match="does not exist for stack"):
+        cf_conn.describe_stack_resource(
+            StackName=stack["StackName"], LogicalResourceId="poop"
```

Comment:
Different name here please - `foo` or `bar` will do 🙂

Reply:
Absolutely! Sorry about that. How about this name https://github.com/spulec/moto/pull/4186/commits/c0c53ba090d71778ad117ec694e18becf3d1042d